source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -463,27 +463,26 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
 
   parsed_alpn_protocols_ = parseAlpnProtocols(config.alpnProtocols());
 
-  // To enumerate the required builtin ciphers, curves, algorithms, and
-  // versions, uncomment '#define LOG_BUILTIN_STAT_NAMES' below, and run
-  //  bazel test //test/extensions/transport_sockets/tls/... --test_output=streamed
-  //      | grep " Builtin ssl." | sort | uniq
-  // #define LOG_BUILTIN_STAT_NAMES
-  //
-  // TODO(#8035): improve tooling to find any other built-ins needed to avoid
-  // contention.
-
-  // Ciphers
-  stat_name_set_->rememberBuiltin("AEAD-AES128-GCM-SHA256");
-  stat_name_set_->rememberBuiltin("ECDHE-ECDSA-AES128-GCM-SHA256");
-  stat_name_set_->rememberBuiltin("ECDHE-RSA-AES128-GCM-SHA256");
-  stat_name_set_->rememberBuiltin("ECDHE-RSA-AES128-SHA");
-  stat_name_set_->rememberBuiltin("ECDHE-RSA-CHACHA20-POLY1305");
+  // Ciphers are configured as a string delimited by ":", with equivalence
+  // groups given as "[alt1|alt2]", and exclusions preceded by "!". For the
+  // purposes of collecting stats -- we want to track all these names. We don't
+  // need to fully parse out the structure of the cipher suites -- just extract
+  // out the names.
+  for (absl::string_view cipher_suite :
```

Comment:
@PiotrSikora is there a better way to do this? E.g. an SSL method?

I'm explicitly including the exclusion-patterns here, though it might not really do any good, because it looks like those are not whole segments from a stat-name perspective.

Reply:
You could get all configured cipher suites using [`SSL_CTX_get_ciphers()`](https://www.openssl.org/docs/manmaster/man3/SSL_get_ciphers.html) and iterate over them using [`SSL_CIPHER_get_name()`](https://www.openssl.org/docs/man1.1.0/man3/SSL_CIPHER_get_name.html) to get the proper names.

This will cover all the cipher suites that can be negotiated in a given context.